src/utils/ICS24Host.sol

```diff
@@ -152,4 +153,18 @@ library ICS24Host {
     function packetAcknowledgementCommitmentBytes32(bytes memory ack) internal pure returns (bytes32) {
         return sha256(ack);
     }
+
+    /// @notice Create a prefixed path
+    /// @dev The path is appended to the last element of the prefix
+    /// @param merklePrefix The prefix
+    /// @param path The path to append
+    /// @return The prefixed path
+    function prefixedPath(bytes[] memory merklePrefix, bytes memory path) internal pure returns (bytes[] memory) {
```

Comment:
I would also love to see a test with this function, even if just to document how to use it correctly. I also think the natspec should mention the expected empty element.

Reply:
The last element is not necessarily expected to be empty. For example, if the counterparty is the cosmwasm implementation, then it would be the address bytes of the contract

Reply:
Ah, and in that case, should the path be appended to that element, not added as another element? Is the full Merkle path ever more than two elements?

Reply:
Yeah, it should be appended. The way IAVL works is weird

Reply:
Not more than 2 elements in wasm or ibc but in theory it is possible

Reply:
Right, I think I understand now. I guess this is how it all works already. And if the add counterparty will be done the same way in ibc-go (["ibc"], [""]), then all of this is OK.

Reply:
Do you wanna add the test @gjermundgaraba?

Reply:
I added a very basic test